### ISSUE TITLE
Can only filter data viewer columns that are in the current visible frame

### ISSUE THREAD

**Issue by adamconroy** [rstudio/rstudio#6147]
### System details

    RStudio Edition : Desktop
    RStudio Version :  1.3
    OS Version      :  Windows 10
    R Version       :  3.5.2

### Steps to reproduce the problem

1) Create a data frame larger than 50 columns
`testData = data.frame(logicalVar = sample(c(T,F), size = 10, replace = T), doubleVars = matrix(rnorm(500), 10))`
`View(testData)`

2) Filter column 1
3) Click the `>>` forward frame button to view columns `2-51`

Note that the filtering is lost

### Describe the problem in detail
We currently do not have any saved state when changing a view frame. The `grid_data` call is entirely stateless so when we change frames that filtering data is lost. Adding state here won't be entirely trivial so I'm making an issue for it.

### Describe the behavior you expected
All set filters are saved between frame pages.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by adamconroy** [comment]
@jmcphers let me know if I should try to squeeze this in for 1.3. It'll be more work than the other issues I'm finishing up so I'm splitting it up into a new potential bucket or work.

**Comment by francisbarton** [comment]
Came here to report this same issue, and glad to see @adamconroy has already submitted such a thorough report. I know you're not supposed to comment on issues just to say "+1", but it is an annoying problem and it would be great to see it fixed when possible. Thanks to all who are working on RStudio.

**Comment by ronblum** [comment]
Note that we now have three duplicate issues:
- #6147, this one—the original, leaving it open
- #7306 
- #7534

**Comment by francisbarton** [comment]
Seems like there was a whole bunch of activity and interest in this and related Data Viewer issues so it feels reasonable to keep this alive?

**Comment by francisbarton** [comment]
By the way, i discovered today that sorting also isn't maintained while paging.

**Comment by luisvalenzuelar** [comment]
Although labelled as an enhancement, it is really a faulty behaviour because it can seriously mislead the user. I think this deserves more urgency than "backlog/later".

**Comment by kellytdunn** [comment]
Agreed, I just discovered that sorting is not retained when scrolling to columns 51 + and am bothered by this.

**Comment by jthomasmock** [comment]
Thanks for your interest! We will be looking at the Data Viewer for general improvements in the future (as seen from #7600)
